tests/behaviour/util.rs

```diff
@@ -20,7 +20,40 @@
  */
 
 use cucumber::gherkin::Step;
+use std::collections::HashMap;
+use typedb_client::concept::*;
 
 pub fn iter_table(step: &Step) -> impl Iterator<Item = &str> {
     step.table().unwrap().rows.iter().flatten().map(String::as_str)
 }
+
+pub fn iter_map_table(step: &Step) -> impl Iterator<Item = HashMap<&String, &String>> {
+    let mut all_rows = step.table().unwrap().rows.iter();
+    let keys = all_rows.next().unwrap();
+    all_rows.map(|row| HashMap::from_iter(keys.iter().zip(row.iter()).collect::<Vec<_>>().into_iter()))
+}
```

Comment:
`HashMap::from_iter(/*...*/.collect::<Vec<_>>().into_iter())` is entirely unnecessary.

`&[]` has a cool method called `.split_first()` that should allow us to streamline this even further:

```suggestion
pub fn iter_map_table(step: &Step) -> impl Iterator<Item = HashMap<&String, &String>> {
    let (keys, rows) = step.table().unwrap().rows.split_first().unwrap();
    rows.iter().map(|row| keys.iter().zip(row).collect())
}
```

Reply:
Fixed